### PR TITLE
Port Verify, Build

### PR DESCRIPTION
**Pull request by dawsontoth**
Verify Port: https://jira.appcelerator.org/browse/TIMOB-18496
Build Changes: https://jira.appcelerator.org/browse/CLI-325
Epic: https://jira.appcelerator.org/browse/TIMOB-18494

We can also do device builds with Tony's latest changes, too, so in theory this also covers:
https://jira.appcelerator.org/browse/CLI-222
https://jira.appcelerator.org/browse/CLI-259

### CODE REVIEW

**Comment by dawsontoth** [comment]
I need to move the AssetCryptImpl.java out of this and in to its proper place before we merge this. Will work on that next.

File: android/src/com/appcelerator/security/AssetCryptImpl.java [hunk 1]
```diff
@@ -0,0 +1,402 @@
...
+            String[] files = _app.getAssets().list(encryptedFn);
+            Arrays.sort(files);
+            MessageDigest mOverallDigest = MessageDigest.getInstance("SHA1");
+            for (String encryptedFile : files) {
+                InputStream in = _app.getAssets().open(encryptedFn + "/" + encryptedFile);
+                MessageDigest mSingleDigest = MessageDigest.getInstance("SHA1");
+                byte[] buf = new byte[8192];
+                while (true) {
+                    int len = in.read(buf);
+                    if (len < 0) {
```
Comment (hieupham007) [inline android/src/com/appcelerator/security/AssetCryptImpl.java]: while (true) seems a little redundant here, perhaps something like this? 
int len = 0
while ( (len = in.read(buf)) != -1) {
     ...
}

File: android/src/com/appcelerator/security/AssetCryptImpl.java [hunk 2]
```diff
@@ -0,0 +1,402 @@
...
+     * Checks that the application is ready for decryption.
+     *
+     * @param app     The app.
+     * @param appInfo The app info.
+     */
+    public AssetCryptImpl(TiApplication app, ITiAppInfo appInfo) {
+        _excepted = false;
+        try {
+            if (DEBUG) {
+                Log.d(TAG, "Starting platform AssetCryptImpl");
```
Comment (hieupham007) [inline android/src/com/appcelerator/security/AssetCryptImpl.java]: I think Log.d(TAG, "", Log.DEBUG_MODE) serves the same purpose

**Comment by dawsontoth** [comment]
@jhaynie @hieupham007 you'll need to force reset your local branches to my new remote branch.

**Comment by ingo** [comment]
@hieupham007 can you please re-review ASAP?

**Comment by hieupham007** [comment]
Code reviewed. Looks good

**Comment by jhaynie** [comment]
I'm out of office at funeral. You guys should merge

**Comment by ingo** [comment]
Approved. Merging.

**Pull request merged**